### PR TITLE
fix: MD023/heading-start-left/header-start-left

### PR DESCRIPTION
**Pull request by nschonni**

### CODE REVIEW

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @nschonni. Left a couple of other extra nits in terms of linting rules plus a suggestion for you to consider for the methods.md topic. Let me know what you think.

File: docs/csharp/methods.md
Comment (nschonni) [reply docs/csharp/methods.md]: The custom anchor was already there, I just unstuck it from the header line. Not sure if it's a VSCode or Markdown issue, but when they're stuck together, the syntax highlighting doesn't work. I agree that they are usually a pain, the only benefit i see to them is that heads can be re-worded without breaking bookmarks

Comment (mairaw) [reply docs/csharp/methods.md]: Yes, I know they were there. So let's leave as-is for now.

File: docs/csharp/methods.md
Comment (mairaw) [inline docs/csharp/methods.md]: do you want to review all the headings in this topic to make them atx style?

Comment (nschonni) [reply docs/csharp/methods.md]: Sure, I'll take a look later, but I think that is a separate MarkdownLint rule. I was just re-enabled the non-passing rules one at a time

Comment (nschonni) [reply docs/csharp/methods.md]: Done. I'll do a separate PR to clean these up after this lands

**Review by BillWagner — COMMENTED** [review]
I looked at this PR today, and I believe all Maira's concerns have been addressed.

I added the changes addressed label, and I'll let Maira approve.

**Review by mairaw — APPROVED** [review]
Looks great! Thanks for doing this @nschonni!

**Pull request merged**
